code/modules/clothing/suits/marine_armor.dm

```diff
@@ -288,51 +288,265 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
+						to_chat(wearer,"<span class='notice'>[src] beeps: Toxicity treatment reservoir replenished.</span>")
+				if(B18_PAIN_CODE)
+					if(B18_pain_cooldown)
+						B18_pain_cooldown = null
+						to_chat(wearer,"<span class='notice'>[src] beeps: Painkiller reservoir replenished.</span>")
+			playsound(src,'sound/effects/refill.ogg', 25, 0, 1)
+
+/obj/item/clothing/suit/storage/marine/specialist/verb/configure_automedic()
+	set name = "Configure Automedic"
+	set category = "B18 Armor"
+	set src in usr
 
-	if(usr.get_active_hand())
-		to_chat(usr, "Your active hand must be empty.")
+	if(!usr.canmove || usr.stat || usr.is_mob_restrained() || usr != wearer )
 		return 0
 
-	to_chat(usr, "You feel a faint hiss and an injector drops into your hand.")
-	var/obj/item/reagent_container/hypospray/autoinjector/tricordrazine/O = new(usr)
-	usr.put_in_active_hand(O)
-	injections--
-	playsound(src,'sound/machines/click.ogg', 15, 1)
+	handle_interface(usr)
+
+/obj/item/clothing/suit/storage/marine/specialist/proc/handle_interface(mob/living/carbon/human/user, flag1)
+	user.set_interaction(src)
+	var/dat = {"<TT>
+	<A href='?src=\ref[src];B18_automed_on=1'>Turn Automed System: [B18_automed_on ? "Off" : "On"]</A><BR>
+	<BR>
+	<B>Use Integrated Health Analyzer:</B><BR>
+	<A href='byond://?src=\ref[src];B18_analyzer=1'>Scan Wearer</A><BR>
+	<BR>
+	<B>Damage Trigger Threshold (Max 150, Min 50):</B><BR>
+	<A href='byond://?src=\ref[src];B18_automed_damage=-50'>-50</A>
+	<A href='byond://?src=\ref[src];B18_automed_damage=-10'>-10</A>
+	<A href='byond://?src=\ref[src];B18_automed_damage=-5'>-5</A>
+	<A href='byond://?src=\ref[src];B18_automed_damage=-1'>-1</A> [B18_automed_damage]
+	<A href='byond://?src=\ref[src];B18_automed_damage=1'>+1</A>
+	<A href='byond://?src=\ref[src];B18_automed_damage=5'>+5</A>
+	<A href='byond://?src=\ref[src];B18_automed_damage=10'>+10</A>
+	<A href='byond://?src=\ref[src];B18_automed_damage=50'>+50</A><BR>
+	<BR>
+	<B>Pain Trigger Threshold (Max 150, Min 50):</B><BR>
+	<A href='byond://?src=\ref[src];B18_automed_pain=-50'>-50</A>
+	<A href='byond://?src=\ref[src];B18_automed_pain=-10'>-10</A>
+	<A href='byond://?src=\ref[src];B18_automed_pain=-5'>-5</A>
+	<A href='byond://?src=\ref[src];B18_automed_pain=-1'>-1</A> [B18_automed_pain]
+	<A href='byond://?src=\ref[src];B18_automed_pain=1'>+1</A>
+	<A href='byond://?src=\ref[src];B18_automed_pain=5'>+5</A>
+	<A href='byond://?src=\ref[src];B18_automed_pain=10'>+10</A>
+	<A href='byond://?src=\ref[src];B18_automed_pain=50'>+50</A><BR>
+
+	</TT>"}
+	user << browse(dat, "window=radio")
+	onclose(user, "radio")
 	return
 
+//Interface for the B18
+/obj/item/clothing/suit/storage/marine/specialist/Topic(href, href_list)
+	//..()
+	if(usr.stat || usr.is_mob_restrained())
+		return
+	if(((istype(usr, /mob/living/carbon/human) && ((!( ticker ) || (ticker && ticker.mode != "monkey")) && usr.contents.Find(src))) || (usr.contents.Find(master) || (in_range(src, usr) && istype(loc, /turf)))))
```

Comment:
Instead of the istype and ticker stuff, use `usr.IsAdvancedToolUser()`, which works similarly similar.